src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/OpenFolderDialog.cs

```diff
@@ -0,0 +1,645 @@
+                FileName = file;
+
+                FileNames = new string[count];
+                for (uint i = 0; i < count; i++)
+                {
+                    resultsArray.GetItemAt(i, out shellItem);
+                    shellItem.GetDisplayName(NativeMethods.SIGDN.DESKTOPABSOLUTEPARSING, out file);
+                    FileNames[i] = file;
+                }
+                return true;
+            }
+            finally
+            {
+                Marshal.ReleaseComObject(openDialogCoClass);
+            }
+        }
+        #endregion
+
+        //---------------------------------------------------
+        //
+        // Public Properties
+        //
+        //---------------------------------------------------
+        #region Public Properties
+        /// <summary>
+        /// Gets or sets the dialog title.
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// Gets the selected file name.
+        /// </summary>
+		public string FileName { get; private set; }
+
+        /// <summary>
+        /// Gets a collection of the selected file names.
+        /// </summary>
+		public string[] FileNames { get; private set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines whether the user can select more than one file.
+        /// </summary>
+        public bool Multiselect { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines whether the file must exist beforehand.
+        /// </summary>
+        public bool EnsureFileExists { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that specifies whether the returned file must be in an existing folder.
+        /// </summary>
+        public bool EnsurePathExists { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines whether to validate file names.
+        /// </summary>
+        public bool EnsureValidNames { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines whether read-only items are returned.
+        /// Default value is true (allow read-only files).
+        /// </summary>
+		public bool EnsureReadOnly { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value that determines the restore directory.
+        /// </summary>
+        public bool RestoreDirectory { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that controls whether to show or hide the list of pinned  places that the user can choose.
+        /// Default value is true.
+        /// </summary>
+        public bool ShowPlacesList { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value that controls whether to show or hide the list of places where the user has recently opened or saved items.
+        /// Default value is true.
+        /// </summary>
+        public bool AddToMruList { get; set; } = true;
```

Comment:
May I know what "Mru" is?

Reply:
most recently used

Reply:
Thank you. But I don’t think abbreviations are suitable here.

Reply:
Names were taken from WindowsAPICodePack